### DISCUSSION TITLE
feature request: a summary field in the yaml config files

### DISCUSSION THREAD

**Discussion by rsuzuki0** [espanso/espanso#1196] [Feature requests and Ideas]
When the expanded text is long or complex, it is useful to have a summary of the expansion. It can simply be inserted as a yaml comment, but you could probably expand the data vocabulary to include "summary" under "trigger" so that they are associated. You may not have to add a function that utilizes it right away... at least, yq or some scripts will be able to read them for now. Obviously, if espanso can add a commandline option summary to show all triggers and their summaries, that would be awesome.

**Comment by federico-terzi** [comment]
@rsuzuki0 Thank you for the feedback, indeed this could be useful and I'm planning to add a `description` field for the upcoming search feature #420 

So stay tuned for that :)

**Comment by NicolasWebDev** [comment]
I just came to ask for the same feature, especially to show that description somehow in the `espanso ls` command, as to see more quickly what the use case of that particular expansion was.

**Comment by smeech** [comment]
Largely fulfilled by the `label:` property, I think.

`espanso match list | grep something` works with labels.
